### PR TITLE
complete fields for diagnostic (Issue #50)

### PR DESCRIPTION
**Pull request by PENGYUXIONG**
- add source, server, and serverity fields for existing diagnostics
- add complete diagnostic method implementention for diagnostic collector

### CODE REVIEW

File: jakarta-eclipse/src/org/jakarta/jdt/diagnostics/FilterDiagnosticsCollector.java [hunk 1]
```diff
@@ -83,15 +94,17 @@ public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnosti
 						Range annotationrange = JDTUtils.toRange(unit, annotationNameRange.getOffset(), annotationNameRange.getLength());
 						
 						if (!isUrlpatternSpecified && !isValueSpecified && !isServletNamesSpecified) {
-							diagnostics.add(new Diagnostic(annotationrange, "The 'urlPatterns' attribute, 'servletNames' attribute or the 'value' attribute of the WebFilter annotation MUST be specified."));
+							diagnostic = new Diagnostic(annotationrange, "The 'urlPatterns' attribute, 'servletNames' attribute or the 'value' attribute of the WebFilter annotation MUST be specified.");
```
Comment (kathrynkodama) [inline jakarta-eclipse/src/org/jakarta/jdt/diagnostics/FilterDiagnosticsCollector.java:97]: Diagnostics should each have a unique code, specific to that diagnostic. If all the diagnostics are from the same specification, in this case servlet, they can have the same source

Comment (PENGYUXIONG) [reply jakarta-eclipse/src/org/jakarta/jdt/diagnostics/FilterDiagnosticsCollector.java:97]: Sorry, I missed this comment before. Now I have updated the PR. Different collectors are using different `CODE` strings now.

File: jakarta-eclipse/src/org/jakarta/jdt/ServletConstants.java [hunk 2]
```diff
@@ -26,4 +28,14 @@
 	public static final String VALUE = "value";
 	public static final String SERVLET_NAMES = "servletNames";
 
+	/* Diagnostics fields constants */
+	public static final String DIAGNOSTIC_SOURCE="jakarta-servlet";
+	public static final String DIAGNOSTIC_CODE = "ExtendHttpServlet";
+	public static final String DIAGNOSTIC_CODE_MISSING_ATTRIBUTE = "CompleteHttpServletAttributes";
+	public static final String DIAGNOSTIC_CODE_DUPLICATE_ATTRIBUTES = "HttpServletAttributesDuplication";
+	public static final String DIAGNOSTIC_CODE_FILTER = "ExtendWebFilter";
```
Comment (kathrynkodama) [inline jakarta-eclipse/src/org/jakarta/jdt/ServletConstants.java:36]: This code should be "ImplementFilter"

File: jakarta-eclipse/src/org/jakarta/jdt/ServletConstants.java [hunk 3]
```diff
@@ -26,4 +28,14 @@
...
+	public static final String DIAGNOSTIC_CODE_FILTER_MISSING_ATTRIBUTE = "CompleteWebFilterAttributes";
+	public static final String DIAGNOSTIC_CODE_FILTER_DUPLICATE_ATTRIBUTES = "WebFilterAttributesDuplication";
+	public static final String DIAGNOSTIC_CODE_LISTENER = "ExtendWebListener";
```
Comment (kathrynkodama) [inline jakarta-eclipse/src/org/jakarta/jdt/ServletConstants.java:39]: This code should be "ImplementListener"

File: jakarta-eclipse/src/org/jakarta/jdt/ServletConstants.java [hunk 4]
```diff
@@ -26,4 +28,14 @@
...
 	public static final String SERVLET_NAMES = "servletNames";
 
+	/* Diagnostics fields constants */
+	public static final String DIAGNOSTIC_SOURCE="jakarta-servlet";
+	public static final String DIAGNOSTIC_CODE = "ExtendHttpServlet";
+	public static final String DIAGNOSTIC_CODE_MISSING_ATTRIBUTE = "CompleteHttpServletAttributes";
+	public static final String DIAGNOSTIC_CODE_DUPLICATE_ATTRIBUTES = "HttpServletAttributesDuplication";
+	public static final String DIAGNOSTIC_CODE_FILTER = "ExtendWebFilter";
+	public static final String DIAGNOSTIC_CODE_FILTER_MISSING_ATTRIBUTE = "CompleteWebFilterAttributes";
+	public static final String DIAGNOSTIC_CODE_FILTER_DUPLICATE_ATTRIBUTES = "WebFilterAttributesDuplication";
```
Comment (kathrynkodama) [inline jakarta-eclipse/src/org/jakarta/jdt/ServletConstants.java:38]: This code should be "InvalidWebFilterAttribute"

File: jakarta-eclipse/src/org/jakarta/jdt/ServletConstants.java [hunk 5]
```diff
@@ -26,4 +28,14 @@
 	public static final String VALUE = "value";
 	public static final String SERVLET_NAMES = "servletNames";
 
+	/* Diagnostics fields constants */
+	public static final String DIAGNOSTIC_SOURCE="jakarta-servlet";
+	public static final String DIAGNOSTIC_CODE = "ExtendHttpServlet";
+	public static final String DIAGNOSTIC_CODE_MISSING_ATTRIBUTE = "CompleteHttpServletAttributes";
+	public static final String DIAGNOSTIC_CODE_DUPLICATE_ATTRIBUTES = "HttpServletAttributesDuplication";
```
Comment (kathrynkodama) [inline jakarta-eclipse/src/org/jakarta/jdt/ServletConstants.java:35]: This code should be "InvalidHttpServletAttribute"

**Review by kathrynkodama — APPROVED** [review]
Looks good to me, just rebase before merging

**Pull request merged**